apis/apiextensions/v1/composition_types.go

```diff
@@ -308,6 +312,14 @@ type ContainerFunction struct {
 	Runner *ContainerFunctionRunner `json:"runner,omitempty"`
 }
 
+// ContainerFunctionImagePullSecret specifies secrets which contain image credentials
+type ContainerFunctionImagePullSecret struct {
+	// Kubernetes secrets name for Composition Function images.
+	// Uses crossplane namespace and service account by default.
+	// +optional
+	Name string `json:"name,omitempty"`
```

Comment:
Is this actually optional? It's the only field in the type. If it's not, please remove the `+optional` marker and the `omitempty` tag. (Same for the revision type.)